SimPEG/directives/directives.py

```diff
@@ -347,16 +348,18 @@ class BaseBetaEstimator(InversionDirective):
     ----------
     beta0_ratio : float
         Desired ratio between data misfit and model objective function at initial beta iteration.
-    seed : int, None
-        Seed used for random sampling.
+    seed : {None, int, array_like[ints], SeedSequence, BitGenerator, Generator}, optional
+        Random seed used for random sampling. It can either be an int,
+        a predefined Numpy random number generator, or any valid input to
+        ``numpy.random.default_rng``.
 
     """
 
     def __init__(
         self,
         beta0_ratio=1.0,
         n_pw_iter=4,
-        seed=None,
+        seed: int | np.random.Generator | None = None,
```

Comment:
Now I need to update these type hints to match the description in the docstrings. Should we keep type hints in this case or just remove them?

Reply:
it is rather large, it would end up being:
```Python
seed: None | int | np.typing.NDArray[np.int_] | np.random.SeedSequence, np.random.BitGenerator, np.random.Generator
```

Reply:
I'm considering defining a [TypeAlias](https://typing.readthedocs.io/en/latest/spec/aliases.html#type-aliases) for the seeds, something like:

```python
RandomSeed: TypeAlias = (
    int
    | np.typing.NDArray[np.int_]
    | np.random.SeedSequence
    | np.random.BitGenerator
    | np.random.Generator
)
```

So the signature of the methods could look like:
```python
    seed: None | RandomSeed = None,
```

We could also define a `SimPEG.typing` module for aliases like this one. I would like to generate a plethora of these, but in some cases they could be useful.

What do you think?